pkg/patterns/addon/pkg/loaders/git.go

```diff
@@ -0,0 +1,130 @@
+package loaders
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-git/go-git/v5"
+	"io/ioutil"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+type GitRepository struct {
+	baseURL string
+	subDir string
+	branch string
+}
+
+var _ Repository = &GitRepository{}
+
+// NewGitRepository constructs an GitRepository
+func NewGitRepository(baseurl string) *GitRepository{
+	repo := parseGitURL(baseurl)
+	return &repo
+}
+
+func (r *GitRepository) LoadChannel(ctx context.Context,  name string) (*Channel, error){
+	if !allowedChannelName(name) {
+		return nil, fmt.Errorf("invalid channel name: %q", name)
+	}
+
+	log := log.Log
+	log.WithValues("baseURL", r.baseURL).Info("loading channel")
+	log.WithValues("baseURL", r.baseURL).Info("cloning git repository")
+
+	if r.subDir != "" {
+		name = r.subDir + "/" + name
+	}
+	b, err := r.readURL(name)
+	if err != nil{
+		log.WithValues("path", name).Error(err, "error reading channel")
+		return nil, err
+	}
+	fmt.Println(string(b))
```

Comment:
Nit: this looks like leftover debug code.  It's OK to log with a very high verbosity (e.g. 4 or 6) instead, but I would probably just delete it - it's most valuable when you're first developing/debugging.